app/models/finance/schedule/npq_leadership.rb

```diff
@@ -12,8 +12,35 @@ class NPQLeadership < NPQ
 
       PERMITTED_COURSE_IDENTIFIERS = IDENTIFIERS
 
-      def self.default
-        find_by(cohort: Cohort.current, schedule_identifier: "npq-leadership-spring")
+      def self.default_for(cohort: Cohort.current)
+        find_by!(cohort:, schedule_identifier: "npq-leadership-spring")
+      end
+
+      def self.schedule_for(cohort: Cohort.current, schedule_date: Date.current)
+        if autumn_schedule_2022?(schedule_date)
```

Comment:
a bit wary of having to hardcode the cohort year here, but I think we might need it for this period and then we can remove 🤔 maybe we can add a comment around this?